addons/mail/static/src/models/activity/activity.js

```diff
@@ -114,6 +114,18 @@ function factory(dependencies) {
                     ];
                 }
             }
+            if ('request_partner_id' in data) {
+                if (!data.request_partner_id) {
+                    data2.creatorPartner = [['unlink-all']];
```

Comment:
Just use 'unlink' for x2one fields, it's more efficient and logical than using the -all version.